File: BackSpring/domain/model/src/main/java/co/com/sofka/okr/c2/model/usuarios/values/Email.java
package co.com.sofka.okr.c2.model.usuarios.values;

import java.util.Objects;

public class Email {
    private final String value;

    public Email(String value) {
        this.value = Objects.requireNonNull(value);
        if(this.value.isEmpty()){
            throw new IllegalArgumentException("The email cannot be empty");
        }

        if (this.value.length() <= 5){
            throw new IllegalArgumentException("The email must be greater than 5 characters");
        }

       if (!value.matches("^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@"+"[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$")){
           throw new IllegalArgumentException("The email is not valid");
       }
    }

    public String getValue() {
        return value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Email email = (Email) o;
        return Objects.equals(value, email.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(value);
    }
}

File: BackNode/domain/user/values/Email.js
function Email(email) {
    if(!email){
        throw new Error("The email cannot be empty")
    }
    if (email.length <= 5){
        throw new Error("The email must be greater than 5 characters");
    }
    const rexEmail = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    if (!rexEmail.test(email)) {
        throw new Error("The email is not valid");
    }
    return {value: email};
}

module.exports = Email;